src/framework/theme/components/select/select.component.ts

```diff
@@ -1180,6 +1180,12 @@ export class NbSelectComponent
         if (event.keyCode === ESCAPE) {
           this.hide();
           this.button.nativeElement.focus();
+        } else if (event.keyCode === ENTER && this.isOptionSearchInputShown) {
+          event.preventDefault();
+          const activeItem = this.keyManager.activeItem;
+          if (activeItem) {
+            this.selectOption(activeItem);
+          }
```

Comment:
maybe we should add the `this.button.nativeElement.focus();` somewhere here as well? It seems that it looses the focus once I select the option with "enter" - so I cannot search-enter-search again